packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -108,57 +109,58 @@ export class BlobManager {
         // When this blob is retrieved, let the container runtime know that the corresponding GC node got updated.
         this.gcNodeUpdated(this.getBlobGCNodePath(blobId));
 
-        return new BlobHandle(
-            `${BlobManager.basePath}/${storageId}`,
-            this.routeContext,
+        return PerformanceEvent.timedExecAsync(
+            this.logger,
+            { eventName: "AttachmentReadBlob", id: storageId },
             async () => {
-                return this.getStorage().readBlob(storageId).catch((error) => {
-                    this.logger.sendErrorEvent(
-                        {
-                            eventName: "AttachmentReadBlobError",
-                            id: storageId,
-                        },
-                        error,
-                    );
-                    throw error;
-                });
+                return new BlobHandle(
+                    `${BlobManager.basePath}/${storageId}`,
+                    this.routeContext,
+                    async () => this.getStorage().readBlob(storageId),
+                );
             },
-        );
+            { start: true, end: true, cancel: "error" });
     }
 
     public async createBlob(blob: ArrayBufferLike): Promise<IFluidHandle<ArrayBufferLike>> {
-        if (this.runtime.attachState === AttachState.Attaching) {
-            // blob upload is not supported in "Attaching" state
-            this.logger.sendTelemetryEvent({ eventName: "CreateBlobWhileAttaching" });
-            await new Promise<void>((resolve) => this.runtime.once("attached", resolve));
-        }
-
-        const response = await this.getStorage().createBlob(blob);
-        const handle = new BlobHandle(
-            `${BlobManager.basePath}/${response.id}`,
-            this.routeContext,
-            // get() should go through BlobManager.getBlob() so handles created while detached can be redirected
-            // to the correct storage id after they are uploaded
-            async () => this.getBlob(response.id).then(async (h) => h.get()),
-        );
-
-        if (this.runtime.attachState === AttachState.Detached) {
-            this.detachedBlobIds.add(response.id);
-            return handle;
-        }
-
-        // Note - server will de-dup blobs, so we might get existing blobId!
-        if (this.pendingBlobIds.has(response.id)) {
-            await this.pendingBlobIds.get(response.id)?.promise;
-        } else if (!this.blobIds.has(response.id)) {
-            this.pendingBlobIds.set(response.id, new Deferred<void>());
-
-            // send blob attach op and wait until we see it to return the handle
-            this.attachBlobCallback(response.id);
-            await this.pendingBlobIds.get(response.id)?.promise;
-        }
-
-        return handle;
+        return PerformanceEvent.timedExecAsync(
```

Comment:
Please reduce scope of code under PerformanceEvent.timedExecAsync() to include only this.getStorage().createBlob(blob);
If you want to make it so broad, I'd at least make sure we capture initial value of this.runtime.attachState and include it as payload (property) of this event, as times will capture vastly different things for attached and non-attached container. But code below also waits for ops, which can take unknown amount of time (given we might lose connection to ordering service). It becomes really hard to account for all of these processes in order to reason about number reported. So it's much simpler to wrap only this.getStorage().createBlob(blob); call with PerformanceEvent.timedExecAsync and leave everything else as is (unmeasured). If we need to, we can always make it more complex / more inclusive later on